deepchem/data/datasets.py

```diff
@@ -1089,7 +1091,9 @@ def merge(datasets: Sequence[Dataset]) -> "NumpyDataset":
           [ids, dataset.ids],
           axis=0,
       )
-
+    k = y.shape[0]
```

Comment:
Can you add a test for this new case? We should have a unit test that would have caught this error